static/src/javascripts/projects/common/modules/ui/cookiesBanner.js

```diff
@@ -0,0 +1,48 @@
+define([
+    'common/utils/config',
+    'common/utils/$',
+    'common/utils/cookies',
+    'common/utils/detect',
+    'common/utils/storage',
+    'common/utils/template',
+    'common/modules/user-prefs',
+    'common/modules/ui/message'
+], function (
+    config,
+    $,
+    cookies,
+    detect,
+    storage,
+    template,
+    userPrefs,
+    Message
+) {
+    /**
+     * Rules:
+     *
+     * EU visitors only
+     * Never seen the cookie message before
+     * Show once only
+     * Show only on FIRST page view
+     * Persist close state
+     */
+    function init() {
+        var EU_COOKIE_MSG = 'GU_EU_MSG',
+            euMessageCookie = cookies.get(EU_COOKIE_MSG);
+
+        if (euMessageCookie && euMessageCookie === 'unseen') {
+            var link = 'https://www.theguardian.com/info/cookies',
+                txt = 'Welcome to the Guardian. This site uses cookies, read our policy <a href="' + link + '" class="cookie-message__link">here</a>',
+                opts = {important: true},
+                cookieLifeDays = 365,
```

Comment:
Could we make this bigger? I don't fancy seeing the banner again once a year… :unamused: